cmd/gossamer/config.go

```diff
@@ -718,10 +712,12 @@ func setDotRPCConfig(ctx *cli.Context, tomlCfg ctoml.RPCConfig, cfg *dot.RPCConf
 	cfg.WSUnsafeExternal = tomlCfg.WSUnsafeExternal
 
 	// check --rpc flag and update node configuration
-	if enabled := ctx.GlobalBool(RPCEnabledFlag.Name); enabled || cfg.Enabled {
-		cfg.Enabled = true
-	} else if ctx.IsSet(RPCEnabledFlag.Name) && !enabled {
-		cfg.Enabled = false
+	rpcFlagIsSet := ctx.IsSet(RPCEnabledFlag.Name)
```

Comment:
Leaving here for reviewer, please double check this change and the one below are okay

Reply:
LGTM 👍

Reply:
Looks good to me.  It improves things.

In exising code, say `cfg.Enabled` is true, but in the flag we pass false, `cfg.Enabled` will be set to true. Which is wrong. `cfg.Enabled` should be set to false since flags takes precedence. 

Proposed code fixes that. If we don't pass the flag, `cfg.Enabled` would stay as it is, but if we do pass it, it will be overridden.